### DISCUSSION TITLE
Errors running defaults

### DISCUSSION THREAD

**Discussion by falcon-ram** [FlorentF9/DeepTemporalClustering#18] [General]
Hello,
I just tried running DeepTemporalClustering using all defaults. i.e. just running
`python DeepTemporalClustering.py`
I ended up with an assertion error:
`assert(timesteps % pool_size == 0)`
On checking the code and using this:
```
print(f"Xshape={X_train.shape}")
print(f"Timesteps = {X_train.shape[1]}")
print(f"PoolSize = {args.pool_size}")
print(f"input_dim={X_train.shape[-1]}")
```
I found out that:
X_train.shape = (930,128,1)
X_train.shape[1] = timesteps = 128
PoolSize = 10 (Default)
input_dim = X_train.shape[-1] = 1

This was using the default CDF data set.
Is this correct or is there some bug in the code. Right now I am thinking that timesteps should actually be X_train.shape[0].

Thank you and regards.

**Comment by sajinpayandath** [comment]
you TIMESTEPS should be a multiple of pool_size

**Reply by FlorentF9** [reply]
That's correct. `pool_size` must divide number of timesteps of your inputs.Yesterday I pushed a small change, to set the default pool_size for CBF to 8 instead of 10. In any case, you should check that the pooling size is adapted to you data.
